jenkinsci/tfs-plugin
# Add support for per-computer TFVC configuration folders

In some environments, the "home" directory is mounted over a network and shared between many computers, including Jenkins servers and their associated build nodes, which eventually leads to corruption of the configuration directory used for TFVC workspaces.  This branch introduces a new, advanced setting that uses a sub-directory for each computer, copying the current configuration directory (if one is found) on the next run.

In other words, turning on the new **Store TFVC configuration in computer-specific folders** option means the TFS SDK will use the following base path for its <code>Cache</code>, <code>Configuration</code> and <code>Logs</code> directories:
`~/.microsoft/Team Foundation/4.0/${HOSTNAME}/`

Manual testing
--------------

1. **Jenkins** > **Manage Jenkins** > **Configure System** and scroll to **TFS/Team Services**
2. Check the new **Store TFVC configuration in computer-specific folders** checkbox and click **Save**:
![image](https://cloud.githubusercontent.com/assets/297515/17760020/5a904806-64ca-11e6-9bcf-829feff881ca.png)
3. Queue a job on MASTER that uses TFVC for its SCM.
    1. Notice that the `4.0` folder now contains a sub-folder named after the server's host name, with copies of the existing folders.
4. Modify the job to execute on a remote node and queue it again.
    1. Notice that the `4.0` folder on the _node_ now contains a sub-folder named after the node's host name.

Mission accomplished!
